commands/cdns.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2018 The Doctl Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"fmt"
+
+	"github.com/digitalocean/doctl"
+	"github.com/digitalocean/doctl/do"
+	"github.com/digitalocean/godo"
+	"github.com/spf13/cobra"
+)
+
+// CDN creates the CDN command
+func CDN() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:   "cdn",
+			Short: "cdn commands",
+			Long:  "cdn is used to access cdn commands",
+		},
+	}
+
+	CmdBuilder(cmd, RunCDNList, "list", "list cdn", Writer,
+		aliasOpt("ls"), displayerType(&cdn{}))
+
+	cmdCDNCreate := CmdBuilder(cmd, RunCDNCreate, "create <cdn-origin>", "create a cdn", Writer,
+		aliasOpt("c"), displayerType(&cdn{}))
+	AddIntFlag(cmdCDNCreate, doctl.ArgCDNTTL, "", 3600, "CDN ttl")
+
+	cmdRunCDNDelete := CmdBuilder(cmd, RunCDNDelete, "delete <cdn-id>", "delete a cdn", Writer,
+		aliasOpt("rm"))
+	AddBoolFlag(cmdRunCDNDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Force cdn delete")
+
+	CmdBuilder(cmd, RunCDNGet, "get <cdn-id>", "get a cdn", Writer, aliasOpt("g"),
+		displayerType(&cdn{}))
+
+	cmdCDNUpdate := CmdBuilder(cmd, RunCDNUpdateTTL, "update <cdn-id>", "update a cdn", Writer,
+		aliasOpt("u"), displayerType(&cdn{}))
+	AddIntFlag(cmdCDNUpdate, doctl.ArgCDNTTL, "", 0, "cdn ttl",
+		requiredOpt())
+
+	cmdCDNFlushCache := CmdBuilder(cmd, RunCDNFlushCache, "flush <cdn-id>", "flush cdn cache", Writer,
+		aliasOpt("fc"))
+	AddStringSliceFlag(cmdCDNFlushCache, doctl.ArgCDNFiles, "", []string{}, "cdn files",
+		requiredOpt())
+
+	return cmd
+}
+
+// RunCDNList returns a list of CDNs.
+func RunCDNList(c *CmdConfig) error {
+	cdns, err := c.CDNs().List()
+	if err != nil {
+		return err
+	}
+
+	return c.Display(&cdn{cdns: cdns})
+}
+
+// RunCDNGet returns an individual CDN.
+func RunCDNGet(c *CmdConfig) error {
+	if len(c.Args) == 0 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	id := c.Args[0]
+	item, err := c.CDNs().Get(id)
+	if err != nil {
+		return err
+	}
+
+	return c.Display(&cdn{cdns: []do.CDN{*item}})
+}
+
+// RunCDNCreate creates a cdn.
+func RunCDNCreate(c *CmdConfig) error {
+	if len(c.Args) == 0 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+
+	origin := c.Args[0]
+
+	ttl, err := c.Doit.GetInt(c.NS, doctl.ArgCDNTTL)
+	if err != nil {
+		return err
+	}
+
+	createCDN := &godo.CDNCreateRequest{
+		Origin: origin,
+		TTL:    uint32(ttl),
```

Comment:
Is there an upper-bound on the ttl server-side? Just wondering what happens here if `ttl` is negative then cast to `uint32`.

Reply:
I just tested it and the endpoint returns a 500 error if the user tries to submit a negative ttl. Most likely because negative numbers cause `uint32` to overflow and return an error. https://play.golang.org/p/VHHvC9BDnpI

I'll add a check to return a `doctl` error if the ttl passed in is negative